src/lib.rs

```diff
@@ -626,6 +626,105 @@ where
         };
     }
 
+    /// Return the number of contiguous clusters. If the next cluster in the sequence isn't contiguous
+    /// (i.e. is fragmented), it returns `1`
+    fn check_contiguous_cluster_count(
+        &self,
+        volume: &Volume,
+        mut cluster: Cluster,
+    ) -> Result<u32, Error<D::Error>> {
+        let mut contiguous_cluster_count = 1u32;
+        let mut next_cluster = match &volume.volume_type {
+            VolumeType::Fat(fat) => match fat.next_cluster(self, cluster) {
+                Ok(cluster) => cluster,
+                Err(e) => match e {
+                    // If this is the last cluster for the file, simply return the same cluster.
+                    Error::EndOfFile => cluster,
+                    _ => panic!(
+                        "Error: traversing the FAT table, accessed free space or a bad cluster"
+                    ),
+                },
+            },
+        };
+        while (next_cluster.0 - cluster.0) == 1 {
+            cluster = next_cluster;
+            next_cluster = match &volume.volume_type {
+                VolumeType::Fat(fat) => match fat.next_cluster(self, cluster) {
+                    Ok(cluster) => cluster,
+                    Err(e) => match e {
+                        Error::EndOfFile => break,
+                        _ => panic!(
+                            "Error: traversing the FAT table, accessed free space or a bad cluster"
+                        ),
+                    },
+                },
+            };
+            contiguous_cluster_count += 1;
+        }
+        Ok(contiguous_cluster_count)
+    }
+
+    /// Read from an open file. It has the same effect as the [`Self::read`] method but reduces `read time`
+    /// by more than 50%, especially in the case of large files (i.e. > 1Mb)
+    ///
+    /// `read_multi` reads multiple contiguous blocks of a file in a single read operation,
+    /// without the extra overhead of additional `data-copying`.
+    ///
+    /// NOTE: The buffer argument must be a multiple of 512 bytes. This impl assumes the underlying
+    /// emmc driver (and consequently the EMMC device) features support for multi-block reads.
```

Comment:
- Changed ~~emmc~~ to `block device` and ~~512 bytes~~ to `block size`. 
- Added a note for what happens when the buffer isn't a multiple of block size.
- For why `read_multi` should be a separate method, `read_multi` depends on `hardware support` for multi-block reads. I guess it would make sense to reflect that in the API and avoid checks to determine support for multi-block read (i.e. hardware dependent code).

Note: by hardware support I mean the actual block device (like an sd-card) not so much the host controller.